DataStructures.Tests/TimelineTests.cs

```diff
@@ -0,0 +1,628 @@
+            };
+            timeline.RemoveValue("TestTime1");
+            Assert.IsTrue(timeline.Count == 4 && !timeline.ContainsValue("TestTime1"));
+        }
+
+        [Test]
+        public static void ToArrayTest()
+        {
+            var timeline = new Timeline<string>
+            {
+                { new DateTime(1995, 1, 1), "TestTime1" },
+                { new DateTime(2000, 1, 1), "TestTime2" },
+                { new DateTime(2005, 1, 1), "TestTime3" },
+                { new DateTime(2010, 1, 1), "TestTime4" },
+                { new DateTime(2015, 1, 1), "TestTime5" }
+            };
+            var array = timeline.ToArray();
+            Assert.IsTrue(timeline.Count == array.Length);
+            int i = 0;
+            foreach (var (time, value) in timeline)
+            {
+                Assert.IsTrue(time == array[i].Time && value == array[i].Value);
+                i++;
+            }
+        }
+
+        [Test]
+        public static void ToListTest()
+        {
+            var timeline = new Timeline<string>
+            {
+                { new DateTime(1995, 1, 1), "TestTime1" },
+                { new DateTime(2000, 1, 1), "TestTime2" },
+                { new DateTime(2005, 1, 1), "TestTime3" },
+                { new DateTime(2010, 1, 1), "TestTime4" },
+                { new DateTime(2015, 1, 1), "TestTime5" }
+            };
+            var list = timeline.ToList();
+            Assert.IsTrue(timeline.Count == list.Count);
+            int i = 0;
+            foreach (var (time, value) in timeline)
+            {
+                Assert.IsTrue(time == list[i].Time && value == list[i].Value);
+                i++;
+            }
+        }
+
+        [Test]
+        public static void ToDictionaryTest()
+        {
+            var timeline = new Timeline<string>
+            {
+                { new DateTime(1995, 1, 1), "TestTime1" },
+                { new DateTime(2000, 1, 1), "TestTime2" },
+                { new DateTime(2005, 1, 1), "TestTime3" },
+                { new DateTime(2010, 1, 1), "TestTime4" },
+                { new DateTime(2015, 1, 1), "TestTime5" }
+            };
+            var dictionary = timeline.ToDictionary();
+            var timelineList = new System.Collections.Generic.List<(DateTime Time, string Value)>();
+            foreach (var pair in timeline)
+                timelineList.Add(pair);
+            var dictionaryList = new System.Collections.Generic.List<(DateTime Time, string Value)>();
+            foreach (var (key, value) in dictionary)
+                dictionaryList.Add((key, value));
+            timelineList.OrderBy(pair => pair.Time);
+            dictionaryList.OrderBy(pair => pair.Time);
+            Assert.IsTrue(timelineList.Count == dictionaryList.Count);
+            for (int i = 0; i < timelineList.Count; i++)
+                Assert.IsTrue(timelineList[i].Time == dictionaryList[i].Time && timelineList[i].Value == timelineList[i].Value);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Correct one of the identical expressions on both sides of operator '=='.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5574777)